### PR TITLE
feat: complete model rendering and file output for JavaScript generator

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR adds both complete and file output for the JS generator.

**Related issue(s)**
Fixes https://github.com/asyncapi/modelina/issues/400

### CODE REVIEW

File: src/generators/javascript/JavaScriptGenerator.ts [hunk 1]
```diff
@@ -27,9 +31,29 @@ export class JavaScriptGenerator extends AbstractGenerator<JavaScriptOptions> {
...
+   * @param model
+   * @param inputModel
+   * @param options
+   */
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async renderCompleteModel(model: CommonModel, inputModel: CommonInputModel, options: JavaScriptRenderCompleteModelOptions): Promise<RenderOutput> {
+    const outputModel = await this.render(model, inputModel);
+    const modelDependencies = model.getNearestDependencies().map((dependencyModelName) => {
+      const formattedDependencyModelName = this.options.namingConvention?.type ? this.options.namingConvention.type(dependencyModelName, { inputModel, model: inputModel.models[String(dependencyModelName)] }) : dependencyModelName;
+      return `const ${formattedDependencyModelName} = require('./${formattedDependencyModelName}');`;
```
Comment (magicmatatjahu) [inline src/generators/javascript/JavaScriptGenerator.ts:47]: Have in mind that someone will want to render models with ESM not CJS module system, so you `JavaScriptRenderCompleteModelOptions` should have option to choose it - by default we can use ESM.

**Review by magicmatatjahu — CHANGES_REQUESTED** [review]

File: src/generators/javascript/JavaScriptGenerator.ts (hunk 1 above)
Comment (jonaslagoni) [reply src/generators/javascript/JavaScriptGenerator.ts:47]: Good catch! Adapting.

**Comment by jonaslagoni** [comment]
@magicmatatjahu I adapted the code to now support different module systems as well as adapted tests.

I also added two examples (why the total changes increased 😅) that show both systems.

File: src/generators/javascript/JavaScriptGenerator.ts [hunk 2]
```diff
@@ -27,9 +31,33 @@ export class JavaScriptGenerator extends AbstractGenerator<JavaScriptOptions> {
...
+      if (options.moduleSystem === 'CJS') {
+        return `const ${formattedDependencyModelName} = require('./${formattedDependencyModelName}');`;
+      }
+      return `import ${formattedDependencyModelName} from './${formattedDependencyModelName}';`;
+    });
+    let modelCode = `export default ${outputModel.result}`;
+    if (options.moduleSystem === 'CJS') {
+      modelCode = `${outputModel.result}
+module.exports = ${outputModel.renderedName};`;
+    }
```
Comment (magicmatatjahu) [inline src/generators/javascript/JavaScriptGenerator.ts:56]: LGTM, but `switch` should be better option here :)

**Review by magicmatatjahu — APPROVED** [review]

File: src/generators/javascript/JavaScriptGenerator.ts (hunk 2 above)
Comment (jonaslagoni) [reply src/generators/javascript/JavaScriptGenerator.ts:56]: Well, I know Sonarcloud don't like small switch statements, so better not 😆

**Pull request merged**